### PR TITLE
Add LinearAlgebra.normalize fallback for scalars (#44835)

### LINKED ISSUE #920: LinearAlgebra.normalize(x) should work for scalars
Currently, this is defined only for arrays, but it seems like we should have
```jl
normalize(x::Number) = x / abs(x)
```
(I just came across this when trying to implement [this algorithm](https://arxiv.org/abs/math-ph/0609050) for random unitary matrices, which is basically `Q * Diagonal(normalize.(diag(R))`, but I discovered that `normalize` didn't work.)

### LINKED ISSUE #44835: LinearAlgebra.normalize(x) should work for scalars
Currently, this is defined only for arrays, but it seems like we should have
```jl
normalize(x::Number) = x / abs(x)
```
(I just came across this when trying to implement [this algorithm](https://arxiv.org/abs/math-ph/0609050) for random unitary matrices, which is basically `Q * Diagonal(normalize.(diag(R))`, but I discovered that `normalize` didn't work.)

### PR DESCRIPTION
**Pull request by mikerouleau**
Adds fallback methods LinearAlgebra.normalize(x), LinearAlgebra.normalize(x,p::Real) and related tests. These are particularly useful when x is a scalar (closes JuliaLang/LinearAlgebra.jl#920).

### CODE REVIEW

**Comment by stevengj** [comment]
Thanks!  Needs an entry in `NEWS.md` since this is a new feature.

File: stdlib/LinearAlgebra/test/generic.jl [hunk 1]
```diff
@@ -375,6 +375,12 @@ end
     @test typeof(normalize([1 2 3; 4 5 6])) == Array{Float64,2}
 end
 
+@testset "normalize for scalars" begin
+    @test normalize(8.0) == 1.0
+    @test normalize(-3.0) == -1.0
```
[the remark this answers is not in the document]
Comment (mikerouleau) [reply stdlib/LinearAlgebra/test/generic.jl:380]: Good suggestion. Just added a commit that fixes this.

File: NEWS.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ Standard library changes
   system image with other BLAS/LAPACK libraries is not
   supported. Instead, it is recommended that the LBT mechanism be used
   for swapping BLAS/LAPACK with vendor provided ones. ([#44360])
+* `normalize(x)` and `normalize(x, p::Real)` fallback methods created,
+  allowing use on any normed vector space, including scalars.
```
Comment (stevengj) [inline NEWS.md:64]:
```suggestion
* `normalize(x, p=2)` now supports any normed vector space `x`, including scalars ([#44925]).
```

**Review by stevengj — APPROVED** [review]

**Comment by dkarrasch** [comment]
The Windows CI processes timed out. Shall we merge this as is or first rebase and give CI another chance?

**Comment by mikerouleau** [comment]
@dkarrasch That's probably not a decision for me, but I'm fine either way. What generally would cause the windows CI to time out in this manner? It doesn't look related to anything in this PR and I don't see anything explicitly failing.

**Pull request merged**